packages/valory/skills/market_creation_manager_abci/rounds.py

```diff
@@ -305,6 +339,114 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+                        + 1,
+                    },
+                )
+                return synchronized_data, Event.ERROR
+
+            if self.most_voted_payload == self.MAX_RETRIES_PAYLOAD:
+                return self.synchronized_data, Event.MAX_RETRIES_REACHED
+
+            if self.most_voted_payload == self.MAX_APPROVED_MARKETS_REACHED_PAYLOAD:
+                return self.synchronized_data, Event.MAX_APPROVED_MARKETS_REACHED
+
+            if self.most_voted_payload == self.SKIP_MARKET_APPROVAL_PAYLOAD:
+                return self.synchronized_data, Event.SKIP_MARKET_APPROVAL
+
+            synchronized_data = self.synchronized_data.update(
+                synchronized_data_class=SynchronizedData,
+                **{
+                    get_name(
+                        SynchronizedData.collected_proposed_markets_data
+                    ): json.loads(self.most_voted_payload),
+                },
+            )
+            return synchronized_data, Event.DONE
+
+        if not self.is_majority_possible(
+            self.collection, self.synchronized_data.nb_participants
+        ):
+            return self.synchronized_data, Event.NO_MAJORITY
+        return None
+
+
+class ApproveMarketsRound(OnlyKeeperSendsRound):
+    """ApproveMarketsRound"""
+
+    ERROR_PAYLOAD = "ERROR_PAYLOAD"
+    MAX_RETRIES_PAYLOAD = "MAX_RETRIES_PAYLOAD"
+
+    payload_class = ApproveMarketsPayload
+    payload_attribute = "content"
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    no_majority_event = Event.NO_MAJORITY
+
+    def end_block(
+        self,
+    ) -> Optional[
+        Tuple[BaseSynchronizedData, Enum]
+    ]:  # pylint: disable=too-many-return-statements
+        """Process the end of the block."""
+        if self.keeper_payload is None:
+            return None
+
+        # Keeper did not send
+        if self.keeper_payload is None:  # pragma: no cover
+            return self.synchronized_data, Event.DID_NOT_SEND
+
+        # API error
+        if (
+            cast(ApproveMarketsPayload, self.keeper_payload).content
+            == self.ERROR_PAYLOAD
+        ):
+            return self.synchronized_data, Event.ERROR
+
+        # Happy path
+        approved_markets_data = json.loads(
+            cast(ApproveMarketsPayload, self.keeper_payload).content
+        )
+
+        approved_markets_count = len(approved_markets_data.get("approved_markets", []))
+
+        synchronized_data = self.synchronized_data.update(
+            synchronized_data_class=SynchronizedData,
+            **{
+                get_name(SynchronizedData.approved_markets_data): approved_markets_data,
+                get_name(SynchronizedData.approved_markets_count): cast(
+                    SynchronizedData, self.synchronized_data
+                ).approved_markets_count
+                + approved_markets_count,
+            },
+        )
+
+        return synchronized_data, Event.DONE
```

Comment:
`approved_markets_count` should be passed via the payload and this snippet can be replaced by calling `super`.

Reply:
https://github.com/valory-xyz/market-creator/pull/71#discussion_r1412121285

Reply:
Fixed